product_variant_configurator/models/pricelist.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+# © 2015 Oihane Crucelaegui - AvanzOSC
+# © 2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# © 2016 ACSONE SA/NV
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3
+
+from openerp import models, fields, tools, api, exceptions, _
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    @api.model
+    def _price_rule_get_multi(self, pricelist, products_by_qty_by_partner):
+        context = self.env.context.copy() or {}
+        if 'price_extra' not in context:
+            return super(ProductPricelist, self)._price_rule_get_multi(
+                pricelist, products_by_qty_by_partner)
+        date = context.get('date') or fields.Date.context_today(self)
+        price_extra = context.get('price_extra')
+
+        products = map(lambda x: x[0], products_by_qty_by_partner)
+        product_uom_obj = self.env['product.uom']
+        price_type_obj = self.env['product.price.type']
+
+        if not products:
+            return {}
+
+        version = False
+        for v in pricelist.version_id:
+            if (((v.date_start is False) or (v.date_start <= date)) and
+                    ((v.date_end is False) or (v.date_end >= date))):
+                version = v
+                break
+        if not version:
+            raise exceptions.Warning(_("At least one pricelist has no active"
+                                       " version !\nPlease create or activate"
+                                       " one."))
+        categ_ids = {}
+        for p in products:
+            categ = p.categ_id
+            while categ:
+                categ_ids[categ.id] = True
+                categ = categ.parent_id
+        categ_ids = categ_ids.keys()
+
+        prod_tmpl_ids = [tmpl.id for tmpl in products]
+
+        # Load all rules
+        cr = self.env.cr
+        cr.execute(
+            'SELECT i.id '
+            'FROM product_pricelist_item AS i '
+            'WHERE (product_tmpl_id IS NULL OR product_tmpl_id = any(%s)) '
+            'AND (product_id IS NULL) '
+            'AND ((categ_id IS NULL) OR (categ_id = any(%s))) '
+            'AND (price_version_id = %s) '
+            'ORDER BY sequence, min_quantity desc',
+            (prod_tmpl_ids, categ_ids, version.id))
+        item_ids = [x[0] for x in cr.fetchall()]
+        items = self.env['product.pricelist.item'].browse(item_ids)
```

Comment:
Maybe we should use `items = self.env['product.pricelist.item'].browse(item_ids).exists()`, to be sure we only read available ids for the current user, because the SQL query doesn't check access rules (`ir.rule`).

@pedrobaeza : What's your opinion about that ?